src/resty/aws/init.lua

```diff
@@ -144,6 +146,8 @@ do
     if #parts < 3 then
       return nil, "not a valid region, only 2 parts; "..region
     end
+
+    parts = tablex.sub(parts, 1, #parts - 1)
```

Comment:
`tablex.sub` will insert a metatable.

```
❯ cat test.lua
local split = require("pl.utils").split
local tablex = require("pl.tablex")
local str = "us-east-1"
local parts = split(str, "-", true)
print(require("inspect").inspect(parts))
parts = tablex.sub(parts, 1, #parts - 1)
print(require("inspect").inspect(parts))

❯ resty test.lua
{ "us", "east", "1" }
{ "us", "east",
  <metatable> = <1>{
    __concat = <function 1>,
    __eq = <function 2>,
    __index = <table 1>,
    __tostring = <function 3>,
    _class = <table 1>,
    _create = <function 4>,
    _init = <function 5>,
    _name = "List",
    append = <function 6>,
    cast = <function 7>,
    catch = <function 8>,
    chop = <function 9>,
    class_of = <function 10>,
    clear = <function 11>,
    clone = <function 12>,
    concat = <function 13>,
    contains = <function 14>,
    count = <function 15>,
    default_map = <function 16>,
    default_map_with = <function 16>,
    extend = <function 17>,
    filter = <function 18>,
    foreach = <function 19>,
    foreachm = <function 20>,
    get = <function 21>,
    index = <function 22>,
    insert = <function 23>,
    is_a = <function 24>,
    iter = <function 25>,
    iterate = <function 26>,
    join = <function 27>,
    len = <function 28>,
    map = <function 29>,
    map2 = <function 30>,
    mapm = <function 31>,
    minmax = <function 32>,
    new = <table 1>,
    partition = <function 33>,
    pop = <function 21>,
    push = <function 34>,
    put = <function 35>,
    range = <function 36>,
    reduce = <function 37>,
    remove = <function 38>,
    remove_value = <function 39>,
    reverse = <function 40>,
    slice = <function 41>,
    slice_assign = <function 42>,
    sort = <function 43>,
    sorted = <function 44>,
    splice = <function 45>,
    split = <function 46>,
    transform = <function 47>,
    <metatable> = {
      __call = <function 48>
    }
  }
}
```

How about this?

```lua
local n = #parts
parts[n] = nil
parts[n-1] = "*"
```

Reply:
Hmm good catch! I defintely don't need this